packages/x-data-grid-pro/src/hooks/features/serverSideData/useGridServerSideCache.ts

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { useGridApiMethod } from '@mui/x-data-grid';
+import { GridPrivateApiPro } from '../../../models/gridApiPro';
+import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
+import { GridGetRowsParams, GridGetRowsResponse, GridServerSideCache } from '../../../models';
+import { GridServerSideCacheApi } from './serverSideInterfaces';
+
+class SimpleServerSideCache {
+  private cache: Record<string, GridGetRowsResponse>;
+
+  constructor() {
+    this.cache = {};
+  }
+
+  static getKey(params: GridGetRowsParams) {
+    return JSON.stringify([
+      params.paginationModel,
+      params.filterModel,
+      params.sortModel,
+      params.groupKeys,
+    ]);
+  }
+
+  set(key: string, value: GridGetRowsResponse) {
+    this.cache[key] = value;
+  }
+
+  get(key: string) {
+    return this.cache[key];
+  }
+
+  clear() {
+    this.cache = {};
+  }
+}
+
+const getDefaultCache = (cacheInstance: SimpleServerSideCache): GridServerSideCache => ({
+  getKey: SimpleServerSideCache.getKey,
+  set: (key: string, value: GridGetRowsResponse) =>
+    cacheInstance.set(key as string, value as GridGetRowsResponse),
+  get: (key: string) => cacheInstance.get(key as string),
+  clear: () => cacheInstance.clear(),
+});
+
+export const useGridServerSideCache = (
```

Comment:
I'm wondering if there wouldn't be a way to expose the cache object directly rather than have to re-implement a trampoline to each cache method on our API?

Reply:
Do you mean that instead of having this hook, we just expose a method that returns the cache object, and both internally and externally, the cache methods (get, set, etc) are used directly?
We'd still need a way to initialize the cache and keep it in sync with the related prop. Don't we?

Then for simple cases like implementing a cache reset button, the users will have to do:

```tsx
const cache = apiRef.current.getDataSourceCache();

<Button onClick={cache.clear}>Reset cache</Button>
```

This imperative cache will also not be reactively bound, so this could become stale, so we might have to make this part of the state.

Reply:
Would it make sense to make the cache part of the data source object?

Reply:
> Would it make sense to make the cache part of the data source object?

I think it should be ok to do so, though my personal preference would be to keep it a separate one not to clutter the data source object much. A separate internal hook for caching also makes this point more relevant.

I'm open to opinions on it though.

Reply:
I think it's beneficial for users to be able to reuse a dataSource for multiple grids without using the same cache.

But yes I did propose to have something like `getDataSourceCache()`. I'm fine with high-level convenience functions like `fetchTopLevelRows`, but I think it's fine to defer the low-level ones to the existing cache object and not on the API directly. I'm fine with expecting users using low-level functions to understand the nuances.